### PR TITLE
OC-12682

### PR DESCRIPTION
**Pull request by slclinica**

### CODE REVIEW

File: src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Autowired
+    private StudyDao studyDao;
+
+    @Autowired
+    private ArchivedDatasetFileDAO archivedDatasetFileDAO;
+
+
+    @ApiOperation(value = "To get latest scheduled extract dataset ids and creation time for the job name at study level", notes = "only work for authorized users with the right access permission")
+    @RequestMapping(value = "/studies/{studyOID}/extractJobs/{jobName}/jobExecutions", method = RequestMethod.GET)
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java:53]: if we are going to update all jobName references to jobUuid, maybe we should change this url to also use jobUuid?

File: src/main/java/core/org/akaza/openclinica/bean/extract/ArchivedDatasetFileBean.java [hunk 2]
```diff
@@ -43,6 +37,9 @@
     private int ownerId;
     private String format;
     private String status;
+    private String jobUuid;
+
+    private String datasetFileUuid;
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/bean/extract/ArchivedDatasetFileBean.java:42]: maybe rename this to jobExecutionUuid?

File: src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        for (ArchivedDatasetFileBean adfb : extracts) {
+            output += " Dataset Id: " + adfb.getDatasetFileUuid() + "  Date Created: " + adfb.getDateCreated() + "\n";
+        }
+
+        return new ResponseEntity<>("Extract files for job name " + jobName + ": \n" + output, HttpStatus.OK);
+    }
+
+
+    @ApiOperation(value = "To get latest scheduled extract dataset ids and creation time for the job name at study level", notes = "only work for authorized users with the right access permission")
+    @RequestMapping(value = "/studies/extractJobs/{fileUuid}/jobExecutions", method = RequestMethod.GET, produces = "application/zip")
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java:93]: change this url to "/extractJobs/jobExecutions/{jobExecutionUuid}/dataset"?

File: src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java (hunk 1 above)
Comment (slclinica) [reply src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java:53]: I think its better the change it when the references to it gets changed to job uuid too in OC-12828. I began to work on it and realized that there's a lot of references to be changed as well in all the job servlets.

**Comment by svadla-oc** [comment]
👍

**Pull request merged**